File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <queue>
#include <bitset>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer = {0, 0};

    while (s != "1") {
        int count = 0;
        for (char c : s) {
            if (c == '0') {
                answer[1]++;
            } else {
                count++;
            }
        }
        s = bitset<64>(count).to_string();
        s = s.substr(s.find('1'));
        answer[0]++;
    }

    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = {0, 0};
        
        while(s.length() > 1) {
            int n = s.length();
            char[] c = s.toCharArray();
            for (char num : c) {
                if(num == '0') {
                    answer[1]++;
                    n--;
                }
            }
            s = Integer.toBinaryString(n);
            answer[0]++;
        }
        return answer;
    }
}